### PR TITLE
python3Packages.term-image: init at 0.7.2

### PR DESCRIPTION
**Pull request by liff**
Add [`term-image`](https://github.com/AnonymouX47/term-image/), needed (optionally) by a new release of [toot](https://github.com/ihabunek/toot/).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 347924` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.term-image</li>
    <li>python311Packages.term-image.dist</li>
    <li>python312Packages.term-image</li>
    <li>python312Packages.term-image.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/term-image/default.nix [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  pname = "term-image";
+  version = "0.7.2";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "AnonymouX47";
+    repo = "term-image";
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/development/python-modules/term-image/default.nix:23]: Helps with branch/tag confusion, especially during Python mass-updates.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/term-image/default.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  optional-dependencies = {
+    urwid = [ urwid ];
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ] ++ optional-dependencies.urwid;
+
+  # test_url needs online access
+  pytestFlagsArray = [ "--ignore=tests/test_image/test_url.py" ];
```
Comment (fabaff) [inline pkgs/development/python-modules/term-image/default.nix:45]:
```suggestion
  disabledTestPaths = [
    # test_url needs online access
    "tests/test_image/test_url.py"
  ];
```

**Comment by liff** [comment]
Thanks for the review, @fabaff, I’ve applied your suggestions.

File: pkgs/development/python-modules/term-image/default.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  urwid,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "term-image";
+  version = "0.7.2";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/term-image/default.nix:18]:
```suggestion
```

has not been available for a long time

File: pkgs/development/python-modules/term-image/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  pythonImportsCheck = [ "term_image" ];
+
+  meta = with lib; {
+    description = "Display images in the terminal with python";
+    homepage = "https://github.com/AnonymouX47/term-image";
+    changelog = "https://github.com/AnonymouX47/term-image/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ liff ];
+  };
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/term-image/default.nix:57]:
```suggestion
  meta = {
    description = "Display images in the terminal with python";
    homepage = "https://github.com/AnonymouX47/term-image";
    changelog = "https://github.com/AnonymouX47/term-image/releases/tag/v${version}";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ liff ];
  };
```

**Comment by liff** [comment]
@Aleksanaa applied your suggestions, thanks.

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build python312Packages.term-image python311Packages.term-image

**Pull request merged**